Models/TermModel.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace Models;
+
+
+use Entities\TermEntity;
+
+class TermModel extends BasicModel
+{
+
+  public function __construct()
+  {
+    parent::__construct();
+  }
+
+  function save($term)
+  {
+    $sql = 'SELECT vid FROM vocabulary WHERE vocabulary LIKE ?';
+    $data = array($term->getVocabulary());
+    $statement = $this->executeStatement($sql, $data);
+    $vocabularyId = $statement->fetchObject()->vid;
+
+    $id = $term->getId();
+    if (isset($id)) {
+      $sql = 'UPDATE terms SET vid=?, name=? WHERE id=?';
+      $data = array($vocabularyId, $term->getName(), $id);
+      $statement = $this->executeStatement($sql, $data);
+    } else {
+      $sql = 'INSERT INTO terms (vid, name) VALUES (?, ?)';
+      $data = array($vocabularyId, $term->getName());
+      $statement = $this->executeStatement($sql, $data);
+    }
+    return $statement;
+
+  }
+
+  /**
+   * @param $id
+   * @return TermEntity
+   */
+  function get($id)
+  {
+
+    $sql = 'SELECT id,vid,name FROM terms WHERE id=?';
+    $data = array($id);
+    $statement = $this->executeStatement($sql, $data);
+    $row = $statement->fetchObject();
+    $term=NULL;
+    if ($row) {
+      $term = new TermEntity(array(
+        "id" => $id,
+        "vocabulary" => $row->vid,
```

Comment:
Properties from TermEntity class should reflect columns from database table.
That means you should have tid(integer), vid(integer) and name(string) as main properties, while vocabulary is optional and it should be a string.